interfaces/cython/cantera/composite.py

```diff
@@ -992,15 +992,33 @@ def set_equivalence_ratio(self, phi, *args, **kwargs):
         to be matched to the `SolutionArray`.
         """
 
-        # broadcast argument shape
+        # If ``phi`` is a scalar, broadcast it to the correct argument shape:
```

Comment:
Thanks for taking a shot at improving this! I think `phi ` can be any shape that's compatible with `self._output_dummy`. I _think_ the key is that `phi` needs to have a lower dimensionality than the other array, but I'm not totally sure. I'd have to read the docs for numpy to see, I can never remember how broadcasting works.

The one-line version in my head is

> Broadcast `phi` to match the output shape

But I don't think that adds much either. It's basically restating the name of the function without any more explanation about _why_ this needs to be done.

I think I'll try to grok the Numpy docs and see if there's an alternative that comes to mind.

Reply:
Fwiw, and if memory serves, I used this function upon suggestion when implementing the set equivalence ratio method. I believe @bryanwweber is describing the behavior correctly: it just makes sure that dimensions are matched. We mostly use 1d where things are simple. For 2d and above, broadcasting adds dimensions based on established rules.

Reply:
> But I don't think that adds much either. It's basically restating the name of the function without any more explanation about why this needs to be done.

You could write:
```
If phi is lower-dimensional than the SolutionArray's shape (for example, a scalar),
broadcast it to have the right number of dimensions.
```